build(docs-infra): update docs examples to latest v9

gkalpak opened the pull request:
This PR cleans-up the docs examples and updates them to v9 (and also updates the code to match what the latest CLI would generate for a new project).

This was originally part of #34374 (and split off for easier reviewing).
Partly addresses: [FW-1609](https://angular-team.atlassian.net/browse/FW-1609)

##
Fixes #34382 and #34383.

mary-poppins commented:
You can preview db5524f at https://pr36015-db5524f.ngbuilds.io/.

mary-poppins commented:
You can preview be3cbce at https://pr36015-be3cbce.ngbuilds.io/.

mary-poppins commented:
You can preview c6c456b at https://pr36015-c6c456b.ngbuilds.io/.

petebacondarwin commented on aio/content/examples/template-expression-operators/src/app/app.component.html:
```diff
@@ -72,8 +72,8 @@ <h3>Using safe navigation operator (<code>?</code>)</h3>
   <!-- #docregion non-null -->
-  <!--No color, no error -->
-  <p *ngIf="item">The item's color is: {{item!.color}}</p>
+  <!-- Assert color is defined, even if according to the `Item` type it could be undefined. -->
```
👍

petebacondarwin commented on aio/tools/examples/shared/package.json:
```diff
@@ -18,74 +18,74 @@
     "core-js": "^2.5.4",
-    "express": "^4.14.1",
-    "rxjs": "^6.5.1",
+    "express": "^4.17.1",
```
Not particularly important for this PR but it is worth noting that `express` should be a `devDependency` right, along with the `@nguniversal/express-engine`...

gkalpak replied:
No, these are added to `dependencies` by the CLI. I guess it makes sense, because these are prod dependencies for your server side (where they are used to prerender the page before sending it to the client).

petebacondarwin reviewed (approved):
LGTM:

I assume that the reason the commits are out of order in the GH UI is because of rebases holding on to old dates... This can be remedied with `git rebase master --ignore-date` but not important.

* commit 10/12 does a bit more than update the tsconfig.json - perhaps add some more explanation to the commit message?
* commit 12/12 !fixup seems like it should be its own commit with its own message, not a fixup...

IgorMinar reviewed (approved):
lgtm - global approval

but as discussed offline this is blocked until the next zone.js patch release is out.

gkalpak commented:
@petebacondarwin 

> This can be remedied with `git rebase master --ignore-date` but not important.

Wow, thx, I didn't know `--ignore-date` works with regular `rebase`. I thought it only works with `--interactive`, which does not work on Windows. But without `--interactive` it works fine :tada:
(Updated my rebase alias :wink:)

> commit 10/12 does a bit more than update the tsconfig.json - perhaps add some more explanation to the commit message?
> commit 12/12 !fixup seems like it should be its own commit with its own message, not a fixup...

In each commit, I make the described change (e.g. update `tsconfig.json`) and all changes required as a result of that to keep the tests passing. For example, updating `tsconfig.json` with some stricter type-checking flags made some template changes necessary. I'll update the commit messages with more info.

[1]: https://github.com/angular/angular/pull/34374/commits/89a37a0202ee3f2131477d697f35c26b0c3dc7a5?file-filters%5B%5D=.ts&hide-deleted-files=true#diff-3e334b5f20890d0e67e19178e1c1cd55R21

mary-poppins commented:
You can preview b6843e7 at https://pr36015-b6843e7.ngbuilds.io/.

gkalpak commented:
Rebased on master, added more info in commit messages (see previous comment) and added a fixup commit (158929e313264ee131833aabfb1e8fd519763d34) to update `boilerplate/elements/package.json` that was recently merged with #33484.

mary-poppins commented:
You can preview 71bd4ff at https://pr36015-71bd4ff.ngbuilds.io/.

mary-poppins commented:
You can preview bc6c64c at https://pr36015-bc6c64c.ngbuilds.io/.

gkalpak committed:
ci(docs-infra): use the regular (non-Ivy) Angular packages in `test_docs_examples_ivy` CI job

The docs examples tests are run both with Ivy turned off and on. When
Ivy is turned on, ngcc is used to convert all dependencies (including
the Angular framework packages to Ivy).

Previously, in order to speed things up, the `test_docs_examples_ivy` CI
job would use Angular packages built with Ivy (from
`dist/packages-dist-ivy-aot`). This however was a deviation from what
happens in real-world applications.

This commit changes the `test_docs_examples_ivy` CI job to always use
the regular Angular packages (as published on npm) and use ngcc to
convert them to Ivy.

Relevant discussion:
https://github.com/angular/angular/pull/35091#discussion_r373775396

gkalpak committed:
build(docs-infra): remove unused `jsconfig.json` file

gkalpak committed:
build(docs-infra): remove unused `test.html` file from `testing` example

This file was from back when the `testing` example was not based on the
Angular CLI and is not used any more.

gkalpak committed:
build(docs-infra): remove unused `ngmodule-faq` and `visual-studio-2015` examples

gkalpak committed:
build(docs-infra): remove obsolete content from `example-config.json` in `setup` example

The `unittesting` property in `example-config.json` is a remnant of the
setup (before `aio/` was moved into the `angular/angular` repo) and has
no effect any more.

gkalpak committed:
build(docs-infra): remove obsolete content from `example-config.json` files

The `build` and `run` properties in `example-config.json` are only used
in example projects of type `systemjs`. In CLI-type projects, `build`
and `run` are ignored.

gkalpak committed:
build(docs-infra): avoid Karma error when loading scripts from code.angularjs.org

In the example apps used in the `ngUpgrade` guide, the AngularJS files
are loaded in `index.html` from `https://code.angularjs.org/`. Latest
browsers prevent loading the scripts from a page running on localhost
due to CORS:

```
Access to script at 'https://code.angularjs.org/1.5.5/angular.js' from origin
'http://localhost:9876' has been blocked by CORS policy: No 'Access-Control-Allow-Origin'
header is present on the requested resource.
```

NOTE:
Loading the files from `code.angularjs.org` is not mentioned in the
guide, therefore it is not needed to mention this config change there.

gkalpak committed:
build(docs-infra): update project structure to cli@9 1/12 (`app.po.ts`)

Update `app.po.ts`.

gkalpak committed:
build(docs-infra): update project structure to cli@9 2/12 (`fixture.debugElement.componentInstance`)

`fixture.debugElement.componentInstance` --> `fixture.componentInstance`

gkalpak committed:
build(docs-infra): update project structure to cli@9 3/12 (`title in a h1 tag`)

Make test description less implementation specific.

gkalpak committed:
build(docs-infra): update project structure to cli@9 4/12 (`test.ts`)

Update `test.ts`.

gkalpak committed:
build(docs-infra): update project structure to cli@9 5/12 (`polyfills.ts`)

Update `polyfills.ts` (`import './zone-flags.ts'` --> `import './zone-flags'`).

gkalpak committed:
build(docs-infra): update project structure to cli@9 6/12 (`index.html`)

Update `index.html` (mostly remove empty line).

gkalpak committed:
build(docs-infra): update project structure to cli@9 7/12 (`.gitignore`, `protractor.conf.js`)

Update configs (`.gitignore`, `protractor.conf.js`).

gkalpak committed:
build(docs-infra): update project structure to cli@9 8/12 (`favicon.ico`)

Update `favicon.ico`.

gkalpak committed:
build(docs-infra): update project structure to cli@9 9/12 (`tslint.json`)

Update `tslint.json`.

gkalpak committed:
build(docs-infra): update project structure to cli@9 10/12 (`tsconfig.json`)

Update `tsconfig[.*].json`.
Also, all make necessary changes to ensure the example apps can be
successfully built with the new, stricter type-checking options.

gkalpak committed:
build(docs-infra): update project structure to cli@9 11/12 (`angular.json`)

Update `angular.json`.

gkalpak committed:
build(docs-infra): update project structure to cli@9 12/12 (dependencies)

Update dependencies.
Also, disable AOT tests for `systemjs`-type examples in Ivy mode,
because our current setup does not work with v9. See #35989 for details.

Fixes #34382
Fixes #34383

gkalpak committed:
build(docs-infra): clean up dependencies and npm scripts in `systemjs` docs examples

Previously, the `package.json` files added as boilerplate to docs
example projects of type `systemjs` were incorrect/outdated. They
contained unused dependencies and npm scripts and omitted used
dependencies.

This is not a big issue, because these examples are not offered as live
StackBlitz examples or downloadable ZIP archives, but having incorrect
`package.json` files is confusing and makes it more complicated to
update these examples.

This commit updates the `package.json` templates for the `systemjs` docs
examples and other configuration files to include used dependencies (and
remove unused ones). It also removes unused npm scripts.

gkalpak committed:
docs(upgrade): remove outdated/non-applicable AOT section from upgrade guide

The AOT compilation section of the `ngUpgrade` guide was
outdated/non-applicable. This commit hides the section to avoid confusion.
Fixing the section and re-introducing it is being tracked in #35989.

gkalpak committed:
fixup! build(docs-infra): update project structure to cli@9 12/12 (dependencies)

gkalpak committed:
fixup! build(docs-infra): update project structure to cli@9 12/12 (dependencies)

mary-poppins commented:
You can preview bb47b04 at https://pr36015-bb47b04.ngbuilds.io/.

gkalpak commented:
Rebased on master and updated `zone.js` to 0.10.3. I also took the opportunity to update `@angular/*` from 9.0.3 to the currently latest 9.0.6.

Marking for merge...
